clang/lib/CIR/Dialect/IR/CIRAttrs.cpp

```diff
@@ -366,12 +378,17 @@ DataMemberPtrAttr::verify(function_ref<InFlightDiagnostic()> emitError, Type ty,
     return failure();
   }
 
-  auto memberIndexValue = memberIndex.getValue().getZExtValue();
-  if (memberIndexValue >= clsStructTy.getNumElements()) {
+  auto memberIndexValue = memberIndex.getValue().getSExtValue();
+  if (memberIndexValue >= static_cast<int64_t>(clsStructTy.getNumElements())) {
```

Comment:
Good catch! Yes we should also check this to prevent someone from mistakenly gives us an incomplete class. In normal CIRGen clang Sema will check this for us, though.

Reply:
But the verifier is also useful to catch bad transformations, so @sitio-couto comments is still valid. Can you please go over all conversations and update with either "applied" or an explanation on why they are not tackled?

Reply:
Updated.